### ISSUE TITLE
Error running Chat API server

### ISSUE THREAD

**Issue by firelightning13** [PygmalionAI/aphrodite-engine#15]
Hello, I was trying my hands on aphrodite engine in my laptop, and I discovered some errors to report @AlpinDale . Here's the error log when I run the chat API server:
```
(aphrodite) [muzz@nobara-laptop aphrodite-engine]$ python -m aphrodite.endpoints.openai.api_server --model models/pythia-70m
INFO 09-18 17:30:15 aphrodite_engine.py:72] Initializing an LLM engine with config: model='models/pythia-70m', tokenizer='models/pythia-70m', tokenizer_mode=auto, revision=None, trust_remote_code=False, dtype=torch.float16, download_dir=None, load_format=auto, tensor_parallel_size=1, seed=0)
Traceback (most recent call last):
  File "/home/muzz/miniconda3/envs/aphrodite/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/home/muzz/miniconda3/envs/aphrodite/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/home/muzz/build/aphrodite-engine/aphrodite/endpoints/openai/api_server.py", line 632, in <module>
    engine = AsyncAphrodite.from_engine_args(engine_args)
  File "/home/muzz/build/aphrodite-engine/aphrodite/engine/async_aphrodite.py", line 451, in from_engine_args
    engine = cls(engine_args.worker_use_ray,
  File "/home/muzz/build/aphrodite-engine/aphrodite/engine/async_aphrodite.py", line 252, in __init__
    self.engine = self._init_engine(*args, **kwargs)
  File "/home/muzz/build/aphrodite-engine/aphrodite/engine/async_aphrodite.py", line 281, in _init_engine
    return engine_class(*args, **kwargs)
  File "/home/muzz/build/aphrodite-engine/aphrodite/engine/aphrodite_engine.py", line 104, in __init__
    self._init_workers(distributed_init_method)
  File "/home/muzz/build/aphrodite-engine/aphrodite/engine/aphrodite_engine.py", line 136, in _init_workers
    self._run_workers(
  File "/home/muzz/build/aphrodite-engine/aphrodite/engine/aphrodite_engine.py", line 695, in _run_workers
    output = executor(*args, **kwargs)
  File "/home/muzz/build/aphrodite-engine/aphrodite/task_handler/worker.py", line 67, in init_model
    self.model = get_model(self.model_config)
  File "/home/muzz/build/aphrodite-engine/aphrodite/modeling/loader.py", line 50, in get_model
    model.load_weights(model_config.model, model_config.download_dir,
  File "/home/muzz/build/aphrodite-engine/aphrodite/modeling/models/gpt_neox.py", line 239, in load_weights
    for name, loaded_weight in hf_model_weights_iterator(
TypeError: hf_model_weights_iterator() takes from 1 to 3 positional arguments but 4 were given
```
Apparently the code `hf_model_weights_iterator` only accepts 3 args, but the code in gpt_neox.py (and llama.py as well) passed 4 args, including `revision`. So I delete that one and it works great, until I found another error:

```
(aphrodite) [muzz@nobara-laptop aphrodite-engine]$ python -m aphrodite.endpoints.openai.api_server --model models/pythia-70m
INFO 09-18 17:19:58 aphrodite_engine.py:72] Initializing an LLM engine with config: model='models/pythia-70m', tokenizer='models/pythia-70m', tokenizer_mode=auto, revision=None, trust_remote_code=False, dtype=torch.float16, download_dir=None, load_format=auto, tensor_parallel_size=1, seed=0)
INFO 09-18 17:20:00 aphrodite_engine.py:201] # GPU blocks: 14147, # CPU blocks: 21845
INFO:     Started server process [25905]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
INFO:     Uvicorn running on http://localhost:8000 (Press CTRL+C to quit)
INFO:     127.0.0.1:58964 - "POST /v1/chat/completions HTTP/1.1" 500 Internal Server Error
ERROR:    Exception in ASGI application
Traceback (most recent call last):
  File "/home/muzz/miniconda3/envs/aphrodite/lib/python3.10/site-packages/uvicorn/protocols/http/h11_impl.py", line 408, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
  File "/home/muzz/miniconda3/envs/aphrodite/lib/python3.10/site-packages/uvicorn/middleware/proxy_headers.py", line 84, in __call__
    return await self.app(scope, receive, send)
  File "/home/muzz/miniconda3/envs/aphrodite/lib/python3.10/site-packages/fastapi/applications.py", line 292, in __call__
    await super().__call__(scope, receive, send)
  File "/home/muzz/miniconda3/envs/aphrodite/lib/python3.10/site-packages/starlette/applications.py", line 122, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/home/muzz/miniconda3/envs/aphrodite/lib/python3.10/site-packages/starlette/middleware/errors.py", line 184, in __call__
    raise exc
  File "/home/muzz/miniconda3/envs/aphrodite/lib/python3.10/site-packages/starlette/middleware/errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
  File "/home/muzz/miniconda3/envs/aphrodite/lib/python3.10/site-packages/starlette/middleware/cors.py", line 83, in __call__
    await self.app(scope, receive, send)
  File "/home/muzz/miniconda3/envs/aphrodite/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 79, in __call__
    raise exc
  File "/home/muzz/miniconda3/envs/aphrodite/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 68, in __call__
    await self.app(scope, receive, sender)
  File "/home/muzz/miniconda3/envs/aphrodite/lib/python3.10/site-packages/fastapi/middleware/asyncexitstack.py", line 20, in __call__
    raise e
  File "/home/muzz/miniconda3/envs/aphrodite/lib/python3.10/site-packages/fastapi/middleware/asyncexitstack.py", line 17, in __call__
    await self.app(scope, receive, send)
  File "/home/muzz/miniconda3/envs/aphrodite/lib/python3.10/site-packages/starlette/routing.py", line 718, in __call__
    await route.handle(scope, receive, send)
  File "/home/muzz/miniconda3/envs/aphrodite/lib/python3.10/site-packages/starlette/routing.py", line 276, in handle
    await self.app(scope, receive, send)
  File "/home/muzz/miniconda3/envs/aphrodite/lib/python3.10/site-packages/starlette/routing.py", line 66, in app
    response = await func(request)
  File "/home/muzz/miniconda3/envs/aphrodite/lib/python3.10/site-packages/fastapi/routing.py", line 273, in app
    raw_response = await run_endpoint_function(
  File "/home/muzz/miniconda3/envs/aphrodite/lib/python3.10/site-packages/fastapi/routing.py", line 190, in run_endpoint_function
    return await dependant.call(**values)
  File "/home/muzz/build/aphrodite-engine/aphrodite/endpoints/openai/api_server.py", line 200, in create_chat_completion
    prompt = await get_gen_prompt(request)
  File "/home/muzz/build/aphrodite-engine/aphrodite/endpoints/openai/api_server.py", line 75, in get_gen_prompt
    raise ModuleNotFoundError(
ModuleNotFoundError: fastchat is not installed. Please install fastchat to use the chat completion and conversation APIs: `$ pip install fschat`
```
I did re-installed fastchat via pip, but it still throws me the error. Apparently, there is one missing dependency which is `accelerate`, so I did `pip install accelerate` and it works now.  Just wanted to report these errors, so you can try to fix them.

**Comment by AlpinDale** [comment]
Hi, thanks for the write-up. 

I admittedly haven't been touched the OpenAI API server in a while. I'll try and do some bug fixes tonight if I can, most of my focus has been on the main API.

**Comment by AlpinDale** [comment]
This issue should be fixed with #19 
Closing as complete.
